### ISSUE TITLE
ACS data more complicated than anticipated by SurveyQuestion - package `haven` - labelled ints

### ISSUE THREAD

**Issue by mitzimorris** [mitzimorris/mrp-kit#108]
working with the ACS data, as downloaded from IPUMS, which provides processing library `[ipumsr](https://cran.r-project.org/web/packages/ipumsr/vignettes/ipums.html)`; this library uses `haven`; the resulting objects are labeled ints,
not factors.

can MRP-KIT handle `haven` data?

<img width="1264" alt="acs_questions_numeric_codes" src="https://user-images.githubusercontent.com/5466612/123527676-5e9e8500-d6af-11eb-8dd5-77f022471991.png">

**Comment by jgabry** [comment]
> Can MRP-KIT detect this and handle this automatically or do we need to provide examples of how to work
> with haven data?

By handle automatically do you mean just automatically converting to factor? I think the `SurveyMap$mapping` method does some automatic factor conversion, but I'm not sure if it would be tripped up by the `haven_labelled` class.

**Comment by lauken13** [comment]
I don't love the Haven labelled data because I find it hard to work with. However, it is pretty common within the survey world to work in software that exports labelled data. Perhaps we could have the SurveyData check for labelled data and auto fill the Survey meta data (questions, answer) if so? 

This might be a good a feature for @Dewi-Amaliah  to tackle! :) @jgabry

**Comment by lauken13** [comment]
So @Dewi-Amaliah the procedure would be:

- Check to see if haven labelled type
- If it is, then take questions from the column labels and responses from the factor labels. 

Assuming @mitzimorris  and @jgabry and @RohanAlexander  are okay with this as a feature!

**Comment by Dewi-Amaliah** [comment]
Hi @lauken13, 
I have added the possible code for this issue [here](https://github.com/lauken13/mrpkit/blob/labelled_handling/mrpkit_haven_labelled.Rmd). Please let me know whether it works and if there's anything I can do to improve this.

**Comment by lauken13** [comment]
Hi folks @jgabry  @RohanAlexander  @Dewi-Amaliah ,

Looking at Dewi's (very good!) code, I think the only way to do this is to use the haven package - Dewi gave it a good shot and so did I, but it seems impossible to convert haven labelled vars without it! What do you all think?

**Comment by RohanAlexander** [comment]
I think using Haven is a good choice here.

**Comment by jgabry** [comment]
That's ok by me. I think we can put it as a "suggested" package (Suggests in the DESCRIPTION file) and just check that it's installed if it's needed. In most cases it's not needed (assuming most people don't have haven labelled data but some do) so we don't need to make it a real dependency (Imports or Depends in the DESCRIPTION file) but can use it when necessary.

**Comment by jgabry** [comment]
@lauken13 @Dewi-Amaliah Here's the code for storing the labels and then removing them from the variables:

```r
labs_1 <- labs_2 <- list()
for (j in 1:ncol(feline_survey_haven)) {
  labs_1[[j]] <- attr(feline_survey_haven[[j]], "label")
  labs_2[[j]] <- attr(feline_survey_haven[[j]], "labels")
  attributes(feline_survey_haven[[j]]) <- NULL
}
```

**Comment by lauken13** [comment]
Couldn't we also do this though?

> responses = list(age = c("level1 measurement" = 1,"level2 measurement" = 2,...)

I think I'm more concerned that the answers can only be either factor levels or the string if not a factor. If this is so, why do the users need to specify the response categories?

**Comment by lauken13** [comment]
Yeah so I think of the responses as acting in much the same way as questions do - i.e., the column name might be "v1" but the question is "What is your age?". Similarly the responses might be factorial but names for ease of working with - e.g., 
"w_nh" but actually the labels/meanings are "white non-hispanic".
